src/client/client.cpp

```diff
@@ -137,7 +139,12 @@ namespace OpcUa
         }
         for (  UserTokenPolicy token : ed.UserIdentifyTokens)
         {
-          if (token.TokenType == UserIdentifyTokenType::ANONYMOUS )
+          if (token.TokenType == UserIdentifyTokenType::USERNAME && has_login)
```

Comment:
Client may be tries to authenticate as root, but server may login as guest... :)